cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Node/LedgerEvents/Governance/ProposeNewConstitution.hs

```diff
@@ -0,0 +1,414 @@
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ head wallets
+    , "--signing-key-file", drepSKeyFp 1
+    , "--signing-key-file", drepSKeyFp 2
+    , "--signing-key-file", drepSKeyFp 3
+    , "--out-file", drepRegTxSignedFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", drepRegTxSignedFp
+    ]
+
+  -- Create constitution proposal
+
+  void $ H.execCli' execConfig
+    [ "conway", "governance", "action", "create-constitution"
+    , "--testnet"
+    , "--governance-action-deposit", show @Int 0 -- TODO: Get this from the node
+    , "--deposit-return-stake-verification-key-file", stakeVkeyFp
+    , "--anchor-url", "https://tinyurl.com/3wrwb2as"
+    , "--anchor-data-hash", proposalAnchorDataHash
+    , "--constitution-url", "https://tinyurl.com/2pahcy6z"
+    , "--constitution-hash", constitutionHash
+    , "--out-file", constitutionActionFp
+    ]
+
+  txbodyFp <- H.note $ work </> "tx.body"
+  txbodySignedFp <- H.note $ work </> "tx.body.signed"
+
+  void $ H.execCli' execConfig
+    [ "conway", "query", "utxo"
+    , "--address", Text.unpack $ paymentKeyInfoAddr $ wallets !! 1
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "utxo-2.json"
+    ]
+
+  utxo2Json <- H.leftFailM . H.readJsonFile $ work </> "utxo-2.json"
+  UTxO utxo2 <- H.noteShowM $ H.noteShowM $ decodeEraUTxO sbe utxo2Json
+  txin2 <- H.noteShow =<< H.headM (Map.keys utxo2)
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "build"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--change-address", Text.unpack $ paymentKeyInfoAddr $ wallets !! 1
+    , "--tx-in", Text.unpack $ renderTxIn txin2
+    , "--tx-out", Text.unpack (paymentKeyInfoAddr (wallets !! 0)) <> "+" <> show @Int 5_000_000
+    , "--proposal-file", constitutionActionFp
+    , "--out-file", txbodyFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "sign"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-body-file", txbodyFp
+    , "--signing-key-file", paymentSKey $ paymentKeyInfoPair $ wallets !! 1
+    , "--out-file", txbodySignedFp
+    ]
+
+  void $ H.execCli' execConfig
+    [ "conway", "transaction", "submit"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--tx-file", txbodySignedFp
+    ]
+
+  txidString <- mconcat . lines <$> H.execCli' execConfig
+    [ "transaction", "txid"
+    , "--tx-file", txbodySignedFp
+    ]
+  !propSubmittedResult
+    <- runExceptT $ handleIOExceptT IOE
+                  $ runExceptT $ foldBlocks
+                      (File $ configurationFile testnetRuntime)
+                      (File socketPath)
+                      FullValidation
+                      [] -- Initial accumulator state
+                      (foldBlocksCheckProposalWasSubmitted (fromString txidString))
+
+  newProposalEvent <- case propSubmittedResult of
```

Comment:
Suggest renaming this `ledgerEvents` since this is a list.

Reply:
:+1: